code/modules/recycling/disposal/holder.dm

```diff
@@ -20,6 +20,8 @@
 	active = FALSE
 	last_pipe = null
 	current_pipe = null
+	for(var/atom/movable/thing in contents)
+		thing.forceMove(get_turf(src))
```

Comment:
Don't do this, Destroy() should never perform side effects

Reply:
If you want to do this, do it in deconstruct()

Reply:
also, `as anything`. Contents will always be `/atom/movable` in Turfs, Objects, and Mobs.

Reply:
This is fixing a rare bug (that came up consistently during testing) where the holder is straight up deleted, which deletes the contents (including mobs) outright. I believe I banished it to "very rare" again by changing how holder merges work, but I think it can still occur if the holder reaches a pipe as it's being deleted, hence this action.

I don't know if the holder object has deconstruct, I'll try when i get home.

Reply:
this fixes a rare bug by adding another bug, Destroy() should *never* perform side effects like this

Reply:
Copy. I've since found the actual cause; the pipe objects (in their *own* `Destroy()` proc) use a locate instead of a for loop to handle stored holder objects, which causes any after the first to be qdel'd with the pipe. So I'll just fix that (and probably move it to the right proc) and remove this code.